lib/app/features/chat/providers/conversation_message_actions_provider.c.dart

```diff
@@ -0,0 +1,238 @@
+) async {
+  final databaseService = ref.watch(conversationsDBServiceProvider);
+  final conversationMessageManagementService =
+      ref.watch(conversationMessageManagementServiceProvider).requireValue;
+
+  final eventSigner = await ref.watch(currentUserNostrEventSignerProvider.future);
+
+  return ConversationMessageActionsService(
+    eventSigner: eventSigner,
+    databaseService: databaseService,
+    env: ref.watch(envProvider.notifier),
+    userPubkey: ref.watch(currentPubkeySelectorProvider),
+    sealService: ref.watch(ionConnectSealServiceProvider),
+    nostrNotifier: ref.watch(nostrNotifierProvider.notifier),
+    wrapService: ref.watch(ionConnectGiftWrapServiceProvider),
+    conversationMessageManagementService: conversationMessageManagementService,
+  );
+}
+
+class ConversationMessageActionsService {
+  ConversationMessageActionsService({
+    required this.env,
+    required this.userPubkey,
+    required this.wrapService,
+    required this.sealService,
+    required this.eventSigner,
+    required this.nostrNotifier,
+    required this.databaseService,
+    required this.conversationMessageManagementService,
+  });
+
+  final Env env;
+  final String? userPubkey;
+  final EventSigner? eventSigner;
+  final NostrNotifier nostrNotifier;
+  final IonConnectSealService sealService;
+  final IonConnectGiftWrapService wrapService;
+  final ConversationsDBService databaseService;
+  final ConversationMessageManagementService conversationMessageManagementService;
+
+  Future<void> deleteMessage(String id) async {
+    await databaseService.markConversationMessageAsDeleted(id);
+  }
+
+  Future<void> bookmarkMessage(List<String> ids, String receiverPubkey) async {
+    final createdAt = DateTime.now().toUtc();
+
+    final encodedRumor = jsonEncode([
+      ids.map((id) => ['e', id]).toList(),
+    ]);
+
+    final tags = [
+      ['d', 'chat_messages'],
+      ['encrypted'],
+    ];
+
+    Logger.log('Encoded rumor $encodedRumor');
+
+    final encryptedRumor = await Nip44.encryptMessage(
+      encodedRumor,
+      eventSigner!.privateKey,
+      receiverPubkey,
+    );
+
+    Logger.log('Encrypted rumor $encryptedRumor');
+
+    final id = EventMessage.calculateEventId(
+      tags: tags,
+      createdAt: createdAt,
+      content: encryptedRumor,
+      publicKey: receiverPubkey,
+      kind: BookmarksEntity.kind,
+    );
+
+    final bookmarkMessage = EventMessage(
+      id: id,
+      tags: tags,
+      createdAt: createdAt,
+      content: encryptedRumor,
+      pubkey: receiverPubkey,
+      kind: BookmarksEntity.kind,
+      sig: null,
+    );
+
+    Logger.log('Bookmark message $bookmarkMessage');
+
+    await nostrNotifier.sendEvent(bookmarkMessage, cache: false);
+  }
+
+  Future<void> sendMessageReaction({
+    required String eventId,
+    required String reaction,
+    required String receiverPubkey,
+  }) async {
+    await _createSealWrapSendReaction(
+      content: reaction,
+      signer: eventSigner!,
```

Comment:
What if `eventSigner` is `null`?

Reply:
Added `throw EventSignerNotFoundException();`